src/onegov/search/integration.py

```diff
@@ -471,12 +473,13 @@ def reindex_model(model: type['Base']) -> None:
                 session.invalidate()
                 session.bind.dispose()
 
-        models = self.get_searchable_models()
-        index_log.info(f'Number of models to be indexed: {len(models)}')
-
         with ThreadPoolExecutor() as executor:
             results = executor.map(
-                reindex_model, (model for model in models)
+                reindex_model, (
+                    model
+                    for base in self.session_manager.bases
+                    for model in searchable_sqlalchemy_models(base)
```

Comment:
Based on the fact that we had duplicates in the search results when using this function we also probably compute the index several times for some models. So same thing here, retrieve the polymorphic base class and remove the duplicate base classes.

You will need to take additional care filtering out the non-searchable instances from the searchable ones. E.g. with `SearchableFile` the polymorphic base class is `File`, but it's not searchable.

Another way to deduplicate without introducing an additional filtering step would be to skip all the searchable models that are a subclass of another searchable model and they both have the same `__tablename__`.

Reply:
Duplicates disappeared once I only search in models that are not base classes of other models